File: QuickWork/QuickWork.Model/Requests/JobPostingUpsertRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace QuickWork.Model.Requests
{
    public class JobPostingUpsertRequest
    {
        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [MaxLength(2000)]
        public string Description { get; set; } = string.Empty;

        [Required]
        public int CategoryId { get; set; }

        [Required]
        public int CityId { get; set; }

        [MaxLength(300)]
        public string? Address { get; set; }

        [Required]
        [Range(0.01, 999999.99)]
        public decimal PaymentAmount { get; set; }

        [Range(0.01, 99.99)]
        public decimal? EstimatedDurationHours { get; set; }

        [Required]
        public DateTime ScheduledDate { get; set; }

        public TimeSpan? ScheduledTimeStart { get; set; }

        public TimeSpan? ScheduledTimeEnd { get; set; }

        [MaxLength(50)]
        public string Status { get; set; } = "Open"; // Open, InProgress, Completed, Cancelled

        public bool IsActive { get; set; } = true;
    }
}

File: QuickWork/UI/QuickWork_Mobile/lib/features/jobs/models/job_posting_upsert_request.dart
/// Request payload for creating a job posting via `POST /JobPostings`.
class JobPostingUpsertRequest {
  const JobPostingUpsertRequest({
    required this.title,
    required this.description,
    required this.categoryId,
    required this.cityId,
    this.address,
    required this.paymentAmount,
    this.estimatedDurationHours,
    required this.scheduledDate,
    this.scheduledTimeStart,
    this.scheduledTimeEnd,
    this.status = 'Open',
    this.isActive = true,
  });

  final String title;
  final String description;
  final int categoryId;
  final int cityId;
  final String? address;
  final double paymentAmount;
  final double? estimatedDurationHours;
  final DateTime scheduledDate;
  final String? scheduledTimeStart;
  final String? scheduledTimeEnd;
  final String status;
  final bool isActive;

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'categoryId': categoryId,
      'cityId': cityId,
      'address': address,
      'paymentAmount': paymentAmount,
      'estimatedDurationHours': estimatedDurationHours,
      'scheduledDate': _formatDate(scheduledDate),
      'scheduledTimeStart': scheduledTimeStart,
      'scheduledTimeEnd': scheduledTimeEnd,
      'status': status,
      'isActive': isActive,
    };
  }

  static String _formatDate(DateTime d) {
    // Backend expects ISO datetime, e.g. 2026-03-01T00:00:00.
    final y = d.year.toString().padLeft(4, '0');
    final mo = d.month.toString().padLeft(2, '0');
    final day = d.day.toString().padLeft(2, '0');
    return '$y-$mo-${day}T00:00:00';
  }
}
